.github/workflows/docker-integration-test.yaml

```diff
@@ -0,0 +1,38 @@
+name: docker-integration-test
+on: [push]
+jobs:
+  docker-integration-test:
+    runs-on: ubuntu-latest
+    services:
+      registry:
+        image: registry:2
+        ports:
+          - 5000:5000
+    steps:
+      - 
+        name: Checkout code
+        uses: actions/checkout@v2
+      - 
+        name: Go build
+        run: CGO_ENABLED=0 GOOS=linux go build .
+      -
+        name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+      -
+        name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+        with:
+          driver-opts: network=host
+      -
+        name: Docker build
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          push: true
+          tags: localhost:5000/grid-intensity-exporter:integration-test
+      -
+        name: Docker run
+        run: docker run -d -p 8000:8000 localhost:5000/grid-intensity-exporter:integration-test
+      - 
+        name: Run integration test
+        run: go test -v -tags=dockerrequired ./integration/test/docker
```

Comment:
The integration test is a regular go test. The `-tags` is so it doesn't run if you call `go test ./...`